### PR TITLE
Faster (and smaller) WASM runtime builds

### PR DESCRIPTION
**Pull request by nazar-pc**
This tackles underlying issue from #12671 (partially) from a different angle by simply not compiling any dependencies in `substrate-wasm-builder` when compiling WASM version of the crate in the first place.

It is a bit awkward that `WasmBuilder` is copy-pasted and not implementing any shared trait with regular one, but I decided it might be acceptable for now.

What this means is that runtime is build faster and smaller.

`target/debug/wbuild/node-template-runtime/target` size:
* before ~1.5G
* after ~0.6G

I didn't bother measuring compile times though.

### CODE REVIEW

File: utils/wasm-builder/Cargo.toml [hunk 1]
```diff
@@ -13,13 +13,28 @@ homepage = "https://substrate.io"
...
+strum = { version = "0.24.1", features = ["derive"], optional = true }
+tempfile = { version = "3.1.0", optional = true }
+toml = { version = "0.5.4", optional = true }
+walkdir = { version = "2.3.2", optional = true }
+sp-maybe-compressed-blob = { version = "4.1.0-dev", path = "../../primitives/maybe-compressed-blob", optional = true }
+filetime = { version = "0.2.16", optional = true }
+wasm-opt = { version = "0.110", optional = true }
+
+[features]
+default = ["std"]
```
Comment (bkchr) [inline utils/wasm-builder/Cargo.toml:28]: Good idea, but way too complicated. Just adding target dependencies with `not(target_arch = "wasm32")` should do it.
Then we also don't need to enable or disable any features. 

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies

**Review by bkchr — CHANGES_REQUESTED** [review]
Great idea! But can be improved.

File: utils/wasm-builder/Cargo.toml (hunk 1 above)
Comment (nazar-pc) [reply utils/wasm-builder/Cargo.toml:28]: I tried that initially, the problem is that `substrate-wasm-builder` itself is never compiled for `wasm32` architecture, it is always compiled to native machine code. The cleanest way to work around that that I found was to add an `std` feature.

I was hoping there is a way to distinguish host from target, but apparently not in this context.

Comment (bkchr) [reply utils/wasm-builder/Cargo.toml:28]: Good point, but you should be able to import wasm-builder conditionally in your downstream project based on the target_arch? Achieving the same without requiring any upstream changes?

Comment (nazar-pc) [reply utils/wasm-builder/Cargo.toml:28]: Almost, `provide_dummy_wasm_binary_if_not_exist` is still necessary for WASM, so I thought having it upstream in a backwards compatible way would be more convenient than reimplementing that function in every runtime.

Comment (bkchr) [reply utils/wasm-builder/Cargo.toml:28]: Why is that required? The wasm binary is only imported on std builds?

Comment (nazar-pc) [reply utils/wasm-builder/Cargo.toml:28]: Indeed. I recall there was something when I was testing, will check again later.

**Comment by nazar-pc** [comment]
Replaced with #12790

**Pull request closed without merging**